File: src/main/java/com/sda/rentalloapp/domain/Car.java
package com.sda.rentalloapp.domain;

import com.sda.rentalloapp.domain.enumeration.BodyType;
import com.sda.rentalloapp.domain.enumeration.EngineType;
import com.sda.rentalloapp.domain.enumeration.FuelType;
import lombok.Data;

@Data
public class Car {
    String model;
    String brand;
    FuelType fuelType;
    EngineType engineType;
    BodyType bodyType;
    int numberOfSeats;
    int trunkCapacityInLitres;
    String combustionPer100Km;
    String bodySerialNumber;
    int pricePerDayInPolishGrosz;
    boolean available;
    int rangeInKm;
    Pictures pictures;
}

File: frontend/src/app/models/car.ts
export interface Pictures {
  mainPictureUrl: string,
  picturesUrls: Array<string>
}

// TODO: align with backend types
export interface Car {
  id: number,
  model: string,
  brand: string,
  fuelType: string,
  engineType: string,
  bodyType: string,
  numberOfSeats: number,
  trunkCapacityInLitres: number,
  combustionPer100Km: string,
  bodySerialNumber: string,
  pricePerDayInPolishGrosz: number,
  available: boolean,
  rangeInKm: number,
  pictures: Pictures
}
